src/Microsoft.AspNetCore.Mvc.RazorPages/DependencyInjection/MvcRazorPagesMvcBuilderExtensions.cs

```diff
@@ -35,5 +35,27 @@ public static class MvcRazorPagesMvcBuilderExtensions
             builder.Services.Configure(setupAction);
             return builder;
         }
+
+        /// <summary>
+        /// Configures Razor Pages to use the specified <paramref name="rootDirectory"/>.
+        /// </summary>
+        /// <param name="builder">The <see cref="IMvcCoreBuilder"/>.</param>
+        /// <param name="rootDirectory">The application relative path to use as the root directory.</param>
+        /// <returns></returns>
+        public static IMvcBuilder WithRazorPagesRoot(this IMvcBuilder builder, string rootDirectory)
```

Comment:
It would be nice to also have an overload that doesn't take a string and sets the site root. Not sure about naming